src/main/java/cz/muni/fi/pa165/bluebat/dao/CustomerDaoImpl.java

```diff
@@ -24,7 +25,7 @@ public void create(Customer customer) {
 
     @Override
     public void delete(Customer customer) {
-        em.remove(customer);
+        em.remove(em.find(Customer.class,customer.getId()));
```

Comment:
I to je možnost, jak to vytvořit, ale pokud mazaný objekt není v DB, tak ti to spadne, protože se to pokusí odstranit null. Proto jsem spíš pro tuto variantu: 
```suggestion
        em.remove(em.contains(customer) ? customer : em.merge(customer));
```

Reply:
bude, psal jsem o tom to chatu, že mi to taky házelo tento problém s detached entitou a to je právě to řešení. Výhoda mého řešení je, že když tam dáš nějakou entitu, která už v db není, tak se nic nestane. Ve tvém řešení to vyhodí výjimku.

Reply:
Ok ide to diki.

Reply:
Nz